Networking/Networking/Remote/AccountRemote.swift

```diff
@@ -122,4 +178,71 @@ private extension AccountRemote {
     enum Constants {
         static let wooCommerceSiteSettingsPath: String = "settings"
     }
+
+    enum ParameterKey {
+        static let name = "name"
+    }
+
+    enum Path {
+        static let settings = "me/settings"
+        static let username = "me/username"
+        static let usernameSuggestions = "users/username/suggestions"
+        static let accountCreation = "users/new"
+    }
+}
+
+/// Necessary data for account credentials.
+public struct CreateAccountResult: Decodable {
+    public let authToken: String
+    public let username: String
+
+    public init(authToken: String, username: String) {
+        self.authToken = authToken
+        self.username = username
+    }
+
+    private enum CodingKeys: String, CodingKey {
+        case authToken = "bearer_token"
+        case username
+    }
+}
+
+/// Possible errors from WPCOM account creation.
+public enum CreateAccountError: Error, Equatable {
+    case emailExists
+    case invalidUsername
+    case invalidEmail
+    case invalidPassword(message: String?)
+    case unexpected(error: DotcomError)
+    case unknown(error: NSError)
+
+    /// Decodable Initializer.
+    ///
+    init(dotcomError error: DotcomError) {
+        if case let .unknown(code, message) = error {
+            switch code {
+            case Constants.emailExists:
+                self = .emailExists
+            case Constants.invalidEmail:
+                self = .invalidEmail
+            case Constants.invalidPassword:
+                self = .invalidPassword(message: message)
```

Comment:
❓ Just curious: can we localize this message?

Reply:
The password could be from multiple reasons (like too short, too simple, or any other requirements) with only one error code, that's why I'm returning the error message here. I'm hoping the error message is localized on the backend with the language in the API request header, I'll test it later when the UI is more ready as a subtask in https://github.com/woocommerce/woocommerce-ios/issues/7879.